torch/distributed/rpc.py

```diff
@@ -0,0 +1,137 @@
+    prior to this method.
+
+    Arguments:
+        name (str): a globally unique name of the local RPC agent. It is
+                    encouraged to use names that conform application context.
+                    (e.g., ``Trainer3``, ``ParameterServer2``, ``Master``,
+                    ``Worker1``, etc.)
+        backend (str): type of RPC backend implementation. Currently,
+                       process group backend ``"pg"`` is the only available
+                       backend implementation. (default: ``"pg"``).
+    """
+    global _agent
+    if backend == 'pg':
+        from . import is_initialized
+
+        if not is_initialized():
+            raise RuntimeError("Using pg RPC backend requires calling "
+                               "init_process_group first.")
+
+        from .distributed_c10d import _default_pg
+        # TODO: issue #23232
+        names = _collect_worker_names(name, _default_pg)
+        name_dict = {names[r] : r for r in range(len(names))}
+        _agent = ProcessGroupAgent(name, name_dict, _default_pg)
+    else:
+        raise RuntimeError("Unrecognized RPC backend ", backend)
+
+
+def rpc_async(to, func, args=None, kwargs=None):
+    r"""
+    Asynchronous RPC. Make an RPC call to run function ``func`` on worker
+    ``to``, and immediately returns a future object of the return value.
+
+    Arguments:
+        to (str): name of the destination worker.
+        func (callable): a builtin function (e.g., ``torch.add``).
+        args (tuple): the argument tuple for the ``func`` invocation.
+        kwargs (dict): is a dictionary of keyword arguments for the ``func``
+                       invocation.
+
+    Returns:
+        A Future object that can be waited on. When completed, the return value
+        of ``func`` on ``args`` and ``kwargs`` can be retrieved from the Future
+        object.
+    """
+    qualified_name = torch.jit._find_builtin(func)
+    if qualified_name is None:
+        raise RuntimeError("unknown builtin function %s." % func)
+
+    args = args if args else ()
+    kwargs = kwargs if kwargs else {}
+    return invoke_rpc(_agent, to, qualified_name, *args, **kwargs)
+
+def rpc_sync(to, func, args=None, kwargs=None):
+    r"""
+    Synchronous RPC. Make an RPC call to run function ``func`` on worker ``to``,
+    and block until the return value is locally available.
+
+    Arguments:
+        to (str): name of the destination worker.
+        func (callable): a builtin function (e.g., ``torch.add``).
+        args (tuple): the argument tuple for the ``func`` invocation.
+        kwargs (dict): is a dictionary of keyword arguments for the ``func``
+                       invocation.
+
+    Returns:
+        The return value of ``func`` on ``args`` and ``kwargs``.
+
+    Example::
+
+        On worker 0:
+        >>> import torch.distributed as dist
+        >>> dist.init_process_group(backend='gloo', ...)
```

Comment:
In general, I would be a bit happier if the example was actually runnable code, that a user could then hack up for their own needs.